### PR TITLE
encode lists with IO.iodata_to_binary/1

### PR DESCRIPTION
**Pull request by idi-ot**
seems to fix https://github.com/phoenixframework/phoenix_live_reload/issues/55

### CODE REVIEW

File: lib/phoenix_live_reload/live_reloader.ex [hunk 1]
```diff
@@ -89,9 +89,13 @@ defmodule Phoenix.LiveReloader do
     end
   end
 
-  defp before_send_inject_reloader(conn, endpoint) do
+  defp before_send_inject_reloader(%{resp_body: resp_body} = conn, endpoint) do
     register_before_send conn, fn conn ->
-      resp_body = to_string(conn.resp_body)
+      resp_body = if is_list(resp_body) do
```
Comment (josevalim) [inline lib/phoenix_live_reload/live_reloader.ex]: We don't need the `if`. We can just always use IO.iodata_to_string/1. Thank you!

**Comment by petermm** [comment]
I posted this on elixirforum https://elixirforum.com/t/phoenix-fails-in-live-reloader-when-response-is-encoded-with-msgpax/5732/10 - good old try/rescue - but maybe clearer and woth considering for an immediate fix, then do the bigger refactor when time allows.

```
  defp before_send_inject_reloader(conn, endpoint) do
    register_before_send conn, fn conn ->
      resp_body = try do
        to_string(conn.resp_body)
      rescue
        _ -> false
      end

      if resp_body && inject?(conn, resp_body) && :code.is_loaded(endpoint) do
```

**Comment by josevalim** [comment]
There is no reason for try/catch. The body needs to be iodata. @michalmuskala has good points though. I will push a fix locally. Thanks @idi-ot!

**Comment by idi-ot** [comment]
I've tried to address @michalmuskala's points, but my approach looks ugly.

**Comment by josevalim** [comment]
@idi-ot it actually looks great! We can clean it up with `with`. I will push a commit on top of yours. :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
Actually, we can't use `with` as it still supports early Elixir versions. So this is as good as it gets! Here is how you could rewrite it though in more recent Elixir versions:

```elixir
register_before_send(conn, fn conn ->
  with true <- html?(conn),
       resp_body = IO.iodata_to_binary(resp_body),
       true <- has_body?(resp_body) and :code.is_loaded(endpoint) do
    [page | rest] = String.split(resp_body, "</body>")
    body = page <> reload_assets_tag(conn) <> Enum.join(["</body>" | rest], "")
    put_in conn.resp_body, body
  else
    _ -> conn
  end
end)
```

**Pull request merged**